File: day1/python/day1_2.py
def calculate_module_mass(mass):
    fuel_mass = mass//3-2
    total = fuel_mass
    while fuel_mass > 0:
        fuel_mass = fuel_mass//3 - 2
        if fuel_mass > 0:
            total += fuel_mass

    return total

with open("../input/input_day1") as fp:
    total_sum = 0
    for line in fp:
        module_mass = int(line.strip())
        total_sum += calculate_module_mass(module_mass)

print(total_sum)

File: day1/go/day1_2.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	var totalSum int
	file, err := os.Open("../input/input_day1")
	if err != nil {
		log.Fatal(err)
	}

	defer file.Close()

	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		v, err := strconv.Atoi(scanner.Text())
		if err != nil {
			log.Fatal(err)
		}
		totalSum += calculateModuleMass(v)
	}

	fmt.Println(totalSum)
}

func calculateModuleMass(mass int) int {
	fuelMass := mass/3 - 2
	total := fuelMass
	for fuelMass > 0 {
		fuelMass = fuelMass/3 - 2
		if fuelMass > 0 {
			total += fuelMass
		}
	}
	return total
}
